core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
>> Didn't get this. Weren't you just arguing above that Spark users will use the standard SparkContext.setLocalProperty way? Going by that I thought this new property handling was mostly relevant for JDBC/ODBC connections.

What about the jobs? For smart connector,  they will use what they were using. Or they can use this new property as well. 

>>Which property you are looking at? The "spark.sql.thriftserver.scheduler.pool" is clearly the only supported way in Spark to set pool over hive-thriftserver: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L401

I am confused because I couldn't find any documentation for this property other than the following which is for Spark 1.2. Why is it that Spark does not document it anymore? I may have to try it to understand it. 

https://spark.apache.org/docs/1.2.0/sql-programming-guide.html#migration-guide-for-shark-user

>> Whats the rationale for not using the "spark.sql.thriftserver.scheduler.pool" name that will make it consistent with existing Spark's hive-thriftserver?

As explained above, two rationale:
1. The property is documented as property for JDBC and ODBC clients. 
2. I am not sure if the property is being used for hive servers anymore.